src/Controller/Component/LoginComponent.php

```diff
@@ -51,6 +51,12 @@ public function handleLogin($errorOnlyPost, $redirectFailure)
         if ($result->isValid()) {
             $user = $request->getAttribute('identity')->getOriginalData();
 
+            if ($service->identifiers()->get('Password')->needsPasswordRehash()) {
+                $user->set('password', $request->getData('password'));
+                $user->setDirty('modified');
```

Comment:
I choose to do this as to not update the modify field since all we are doing is rehashing the password.